app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/services/notifications/FirebaseNotificationService.java

```diff
@@ -29,19 +43,46 @@ public Task<Void> subscribeToUserMealPosts(String uid) {
     }
     @Override
     public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
+        createNotificationChannel();
+
         String mealId = remoteMessage.getData().get("mealId");
-        System.out.println(mealId);
         String vendorID = remoteMessage.getData().get("vendorID");
-        System.out.println(vendorID);
-        new Handler(Looper.getMainLooper()).post(() -> {
-            Toast toast = Toast.makeText(getApplicationContext(),
-                    "New meal posted by "+vendorID+": "+mealId,
-                    Toast.LENGTH_LONG);
+        AppContainer appContainer = ((NeighborFoodApplication) this.getApplication()).getAppContainer();
+        /*
+        I couldn't fetch the data from the database that's why I had to remove the following two lines
+
+        User vendor = appContainer.getUserRepo().getUserById(vendorID).getResult();
 
-            toast.show();
+        AtomicReference<Meal> meal = new AtomicReference<Meal>();
+        appContainer.getMealRepo().getMealById(mealId).addOnSuccessListener(result->{
+          meal.set(result);
         });
+        */
+        User vendor = new User("id1", "email@epfl.ch", "gordon"," ramsey");
+        Meal meal = new Meal("paella","delicious paella","delicious paella",R.drawable.fondue);
+
+        //create intent when clicking on notification which redirects to the mealActivity
+        Intent intent = new Intent(this, MealActivity.class);
+        intent.putExtra("name", meal.getName());
+        intent.putExtra("shortDes", meal.getShortDescription());
+        intent.putExtra("longDes", meal.getLongDescription());
+        intent.putExtra("imageid", meal.getImageId());
```

Comment:
A refactoring of the meal fragment activity is needed, as for the intent, we will pass only the id of the meal, so that we can fetch the info from the repository

Reply:
Yes i noticed that when i was creating the intent for the meal activity.. passing a picture id as extra in the intent wont work and I had to change the value of imageId with a dummy one for the implementation since it's the only solution for the current implementation of meal Activity